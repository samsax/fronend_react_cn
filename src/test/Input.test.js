import React, { Component } from 'react';
import {shallow} from 'enzyme';
import Input from '../components/Input';
import '../setupTests'

describe("<Input/>", () => {

    it("Simple render component", () => {
        const component = shallow(
            <Input 
                name="prueba"
                label="Prueba:"
                type="text"
                value=""
                onChange={()=>{}}
            />
        )

        expect(component).toMatchSnapshot()


    })

    it("Check if propd pass correctly", () => {
        const component = shallow(
            <Input 
                name="prueba"
                label="Prueba:"
                type="text"
                value=""
                onChange={()=>{}}
            />
        )
        expect(component.find('label').text()).toBe("Prueba:")
        expect(component.find('input').filterWhere(
            (item) => item.prop('name')==="prueba")
            ).toHaveLength(1)
        expect(component.find('input').filterWhere(
            (item) => item.prop('type')==="text")
            ).toHaveLength(1)
    })

})