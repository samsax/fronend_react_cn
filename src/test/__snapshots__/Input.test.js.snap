// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input/> Simple render component 1`] = `
<Fragment>
  <label
    htmlFor="prueba"
  >
    Prueba:
  </label>
  <input
    className="form-control"
    id="prueba"
    name="prueba"
    onChange={[Function]}
    type="text"
    value=""
  />
</Fragment>
`;
